bin/oneoff/back_populate_user_levels_level_source_id

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env ruby
+#
+# This script populates the `level_source_id` column of the `user_levels` table,
+# if not already present, using the most recent activities `level_source_id`.
+#
+# This is being done as a one-off script so as to be able to control the timing
+# of its execution in production. A migration will follow after the column is
+# populated in production so as to assure the column is populated in all
+# environments.
+
+require_relative '../../dashboard/config/environment'
+
+if ARGV.length != 4
+  puts 'Usage: ./bin/oneoff/back_populate_user_levels_level_source_id start_index end_index batch_size delay_sec'
+  exit 1
+end
+
+start_index = ARGV[0].to_i
+end_index = ARGV[1].to_i
+batch_size = ARGV[2].to_i
+delay_sec = ARGV[3].to_f
+
+until start_index > end_index
+  UserLevel.
+    where(id: start_index..(start_index + batch_size)).
+    where(level_source_id: nil).each do |user_level|
+
+    activity = Activity.
+      where(user_id: user_level.user_id, level_id: user_level.level_id).last
```

Comment:
Ah right, that's is the weird quirk of `last_attempt` vs. dot color.  A level can load a last attempt from a different script even though the dot (`UserLevel`) is 'not_tried'.